components/benefit_sponsors/app/models/benefit_sponsors/concerns/employer_profile_concern.rb

```diff
@@ -183,19 +183,21 @@ def published_benefit_application
       def billing_benefit_application(billing_date=nil)
         billing_report_date = billing_date.is_a?(Date) ? billing_date : TimeKeeper.date_of_record.next_month
         valid_applications = benefit_applications.non_draft.non_imported
-        application = valid_applications.effective_period_cover(billing_report_date).first
-        if billing_date.blank? && application.blank?
 
-          application = valid_applications.future_effective_date(billing_report_date).open_enrollment_period_cover.first
-          return application, application.start_on if application.present?
+        if billing_date.present?
```

Comment:
updated method to return correct benefit application for billing period on page load based on billing report date.